### DISCUSSION TITLE
Installing a Second SSD in the Lenovo ThinkPad T480s

### DISCUSSION THREAD

**Discussion by giscus** [benmepham/benmepham.github.io#7] [Blog Comments]
# Installing a Second SSD in the Lenovo ThinkPad T480s

tl;dr: The T480s supports booting from a second NVMe SSD in the WWAN slot.

https://bjm.me.uk/blog/thinkpad-t480s-ssd/

**Comment by timlod** [comment]
Hi Ben,
I got the same SSD for my T480s after reading your article and successfully cloned the original SSD with Macrium Reflect (I want to dualboot Windows and Linux). 
However, the second SSD doesn’t show up as a boot device in my BIOS. 
Now I found some Reddit posts and another blog post about the T480 which suggest booting from the 2242 only works on the T480, not the T480s. 
Do you perhaps have any further clues about this?

**Reply by timlod** [reply]
I made it work - in case anyone else is looking for this, here's what was the problem on my end:
Windows was installed in Legacy BIOS mode, not UEFI. If the 2242 SSD has an installation using a legacy BIOS, it won't be recognized in the bootloader.
Once I updated Windows to UEFI and re-cloned the installation onto the 2242, it was recognized in the bootloader (without using grub, etc.).
So, yes, booting from a 2242 M.2 SSD is possible on a T480s!

**Reply by benmepham** [reply]
Thanks for sharing, pleased you got it working :slightly_smiling_face:

**Comment by muttleytm** [comment]
I just got a 480s and am looking into doing this as well. Actually the laptop came with Windows installed and I was planning on just moving that drive to an external USB case, but I learned that wouldn't work. I think I remember a few years back reading this sort of thing was in the works and maybe that is why I put off replacing my system.
I was told that since slot the card that windows will be mounted in is related to wifi that wifi might no longer work. I don't know if that means at all or only when booted to the drive in that slot, that is if this is even correct.  It woudn't be a deal breaker for me, but I figured I'd ask about it first.

**Reply by timlod** [reply]
Not sure I understand you correctly, but the small nvme goes in the slot where there would otherwise be the GSM component to have 4G with a SIM card. 
I'm running this setup daily now and have no issues with Wi-Fi in neither Linux nor Windows (or anything for that matter, it works well)!

**Comment by muttleytm** [comment]
OK, good to hear. I'm probably not describing the slot very well since I'm
not sure what it is. I know there are antenna wires in it. I also know is
the only spare one. I had asked someone with more familiarity with this
computer than me, but he was not familiar with using this space to but a
second drive and was sort of guessing.

That's good to hear, particularly because I already found and ordered a
used 256G one. So it sounds like it is just a matter of cloning the windows
drive while it is installed in the computer to this drive while it is
installed as well and that involves cloning to a smaller drive. I'm not
familiar with the app that was mentioned but I did find an article
explaining it can be done and how. I thought I'd wait till I got the stuff
before I read it though.

I don't intend to actually use this drive much. I saw a thread on reddit
from someone who says they had a setup like this and had for a while and
said he advised against it long term as it caused heat issues. In my case
most of the time it wouldn't even by using it. So it doesn't seem that it
is idle it would generate much heat. That discussion also mentioned it
would only use one pci channel or lane for data which I suppose would make
it slow, but I don't know that this, if correct, would generate any heat
and it would only be when it is in use as well. But like most things, I
only know the tip of the iceberg at best.

Thanks for the reply.

**Comment by muttleytm** [comment]
I received a 256G m.2 2245 NVme  today. It physically fit.

However, when I went to boot up the laptop didn't boot, not did any of the little indicator lights light up either. I took it out and and reset things as I'd disabled the internal battery with bios, reset using a pin, etc. and started it up and it and it started normally. So I tried a few more times and it not only doesn't boot up, nothing at all lights up. So I'm thinking something is wrong with it or is the wrong type somehow.

There are a lot of numbers on the back of it. I do see m.2 and nvme on it. I didn't see 2242 or any of the other sizes of this type I'm aware of.

I did check what bios the windows I'll eventually clone was installed with and it was uefi and I saw the bios was set to just run that as well, but I think that wouldn't come into play until I tried to access it or reat or write to it.

I got it used and it was pretty cheap although it was supposed to be tested and was working. So maybe it simply doesn't work, but I really know little about these sorts of drives so maybe there is something else it might be.

Any ideas?

**Reply by timlod** [reply]
I suggest you read the original article again and also my comment above (https://github.com/benmepham/benmepham.github.io/discussions/7#discussioncomment-4120259). It's not clear from your reply whether you get into the BIOS at all when the new card is plugged in. If nothing works once you connect the card, I have no clue. May be something wrong with the unit. 

As I stated before at first I wasn't able to boot from the extra NVME until I updated Windows to UEFI. If that's what you're seeing it could be your issue. If it's something else, I'm afraid I can't help you with that. Good success!

**Comment by muttleytm** [comment]
I only messed with the bios to disconnect the battery as that was the
suggested method. I've always had removable batteries and though I don't
remember doing so I would have just removed them when I changed anything
inside like memory or a ssd.

I did check to see the version of windows I intend to clone and it was
installed with uefi, and the bios was also set for this. So that looks like
that wouldn't be a problem and also it doesn't seem like it would matter
that early in the startup either and I'm not trying to boot to that ssd
either. I'm just trying to boot from the regular boot drive.

At this point I'm thinking the ssd has something wrong with it or doesn't
work.

It is a samsung and only has one slot and only has an m key. The connector
it fits into only has one slot to fit into. I've noticed that some that are
said to work have a b and an m key. So now as I am looking at sites with
reviews or questions I see that all the ones with only m keys, event though
they are nvme, pcie,  2242, etc (I also read about nvme cards and drives
now) in the reviews or questions either the seller says they don't work
with the t480s or the buyers say they didn't work. I sort of thought since
it only had the m slot it would surely work.

I've sent a note to the seller. I've also been looking for others to see if
I can find something with a good price and not take several weeks to
arrive. So at this point I'm waiting to see what the seller says.

I was using a

**Comment by muttleytm** [comment]
Got the right M.2 drive yesterday and things worked fine.

**Reply by benmepham** [reply]
I'm pleased you got it sorted in the end, NVMe drive compatibility can certainly be a nightmare. Out of interest what drive did you go with?

And thanks @timlod for the advice :)

**Comment by muttleytm** [comment]
It was a Toshiba 500 GB which is the same capacity as the drive with windows on it that came with the laptop. The process looked pretty simple even if I had to shrink partitions, but in this case I didn't have to do anything
What was odd is that some have one slot the M key I think it is and other have two, an M key and a B key. So the first one physically fit the laptop fine. The computer only had one recess for the slot to fit into. So while I wasn't sure if the one with two slots would work, I figured for sure the first one I tried would work. 
It isn't the first time I've been wrong.
The seller of the first one, a 256GB told me to return it, I did and I got a refund. So it all ended well. It just took a while.

Thanks for posting the original post about this and replying.

**Comment by Karido** [comment]
Hey guys, how is your longterm review about installing the NVMe and using it as a dual boot? Did you experience any battery damage or overheating? Or is everything running fine? Thanks in advance for the feedback!

**Reply by timlod** [reply]
No problems on my end. I just upgraded my laptop though as I needed a stronger CPU for some creative software, so I wouldn't have a much longer term outlook. But 8 months of heavy use in both Windows and Linux - great setup!

**Comment by danielknlai** [comment]
Hello,
I've got a T480s for home. But the original disk capacity of 256G is too small. Can I use the WWAN slot to install a second large capacity SSD just for data storage purpose (that is to say, it will be a complementary data disk without any booting possibility) ? Will the system recognize this second SSD that I use to store photos and documents ?
Many thanks in advance
Daniel

**Reply by benmepham** [reply]
Hi Daniel,
Yes this should work absolutely fine as long as the second SSD is of the 2242 NVMe variety.
Thanks,
Ben

**Comment by danielknlai** [comment]
Many many thanks Ben.
One more stupid question. Should there be any configuration parameters to be modified before it works. Or the bios shall detect the changes ?
Thanks again
Daniel

**Reply by benmepham** [reply]
Hi, sorry for the delay. No configuration changes should be necessary, it should just detect the change.
